frontend/src/shared/components/FormDialog.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from "react";
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogProps,
+  DialogTitle,
+  DialogContentText,
+  TextField,
+  Grid,
+} from "@mui/material";
+import { SummaryEntry, DialogField } from "shared/types";
+
+interface FormDialogProps extends DialogProps {
+  title: string;
+  mainContent: string;
+  fields: DialogField[];
+  summaryContent: SummaryEntry[];
+  onConfirm: () => void;
+}
+
+export default function FormDialog({
+  title,
+  mainContent,
+  summaryContent,
+  fields,
+  onConfirm,
+  ...props
+}: FormDialogProps) {
+  const handleClose = () => {
+    if (props.onClose) props.onClose({}, "backdropClick");
+  };
+
+  return (
+    <>
+      <Dialog {...props}>
+        <DialogTitle>Subscribe</DialogTitle>
```

Comment:
and hit the bell button hehe
tak na serio, dlaczego to się tak nazywa? xd

Reply:
No nie zmieniłem jeszcze, bo w sumie nwm jaki tytuł dać, ale to raczej dam zmienną